### ISSUE TITLE
variable declaration is removed unexpectedly

### ISSUE THREAD

**Issue by miserylee** [terser/terser#1351]
**Bug report or Feature request?**

Bug report

**Version (complete output of `terser -V` or specific git commit)**

terser 5.16.4

**Complete CLI command or `minify()` options used**

**`terser` input**

```javascript
// test-1.js

(function (factory) {
    factory(exports);
})((function (exports) {
    var a = function () {
        var b = 0;
        var _this = {};
        _this.fn = function () {
            return b;
        }
        _this.fn2 = function () {
            b += 1;
        }
        return _this;
    }
    exports.a = a;
}));

```

Command: `terser ./test-1.js --compress`

**`terser` output or error**

```javascript
!function(exports){var a=function(){var _this={fn:function(){return b},fn2:function(){b+=1}};return _this};exports.a=a}(exports);
```

**Expected result**
This result is produced by terser 5.16.3
```javascript
!function(exports){var a=function(){var b=0,_this={fn:function(){return b},fn2:function(){b+=1}};return _this};exports.a=a}(exports);
```

**Comment by fabiosantoscode** [comment]
Thanks for reporting! I suspect [this change](https://github.com/terser/terser/commit/8dc5fc298d057751acc56cdc84c5cd353be180ee#diff-9aa82f0ed674e050695a7422b1cd56d43ce47e6953688a16a003bf49c3481622R108). I'll have a look tomorrow

**Comment by reey** [comment]
@fabiosantoscode Seems like https://github.com/terser/terser/commit/8dc5fc298d057751acc56cdc84c5cd353be180ee broke this.. Can we revert this commit?

**Comment by fabiosantoscode** [comment]
No problem @reey! Thanks for investigating

**Comment by fabiosantoscode** [comment]
Released in 5.16.5
